### PR TITLE
Add option to scroll to new track on song change

### LINKED ISSUE #798: [SUGGESTION] Add a 'Scroll to playing track' button
Vanilla Music Version : 1.0.62
- Android Version: 4.4.4
- Phone vendor / model: Galaxy S5
- Application was downloaded from :
  - [X] Play Store
  - [ ] F-Droid
  - [ ] Other (Nightly Build)
- Did this work in a previous version?
  - [ ] Yes
  - [X] No
  - [ ] NA / Don't know.

## Description

The current scroll to track function is very useful to not lose focus of the playing song and the tracks surrounding it.

However we may scroll into the library and lose the track location's.

So I thought about an option located inside the three dot menu on the queue (next to the search button, below sorting).

Many times, I had to use the search to find the playing song.

That option would be welcomed.

### PR DESCRIPTION
**Pull request by minus7**
Adds a new setting to scroll the queue to the new song after a song
change. The new song will be at the top of the view.

Kind of fixes #798

### CODE REVIEW

File: app/src/main/java/ch/blinkenlights/android/vanilla/PreferencesActivity.java [hunk 1]
```diff
@@ -194,6 +194,15 @@ public void onCreate(Bundle savedInstanceState)
...
 
+	public static class QueueFragment extends PreferenceFragment {
+		@Override
+		public void onCreate(Bundle savedInstanceState)
+		{
+			super.onCreate(savedInstanceState);
+			addPreferencesFromResource(R.xml.preference_queue);
+		}
+	}
+
```
Comment (adrian-bl) [inline app/src/main/java/ch/blinkenlights/android/vanilla/PreferencesActivity.java]: Having a queue preferences screen with only a single option will look strange: I'd rather stuff that into the misc options (or the library screen, next to the existing 'jump to song' option)

File: app/src/main/java/ch/blinkenlights/android/vanilla/ShowQueueFragment.java [hunk 2]
```diff
@@ -237,7 +238,10 @@ public void setSong(long uptime, Song song) {
 	 */
 	public void onTimelineChanged() {
 		if (PlaybackService.hasInstance()) {
-			refreshSongQueueList(false);
+			SharedPreferences sharedPrefs = PlaybackService.getSettings(getActivity());
```
Comment (adrian-bl) [inline app/src/main/java/ch/blinkenlights/android/vanilla/ShowQueueFragment.java]: Better to pass the context of the full app: getActivity().getApplicationContext()

File: app/src/main/java/ch/blinkenlights/android/vanilla/PreferencesActivity.java (hunk 1 above)
Comment (minus7) [reply app/src/main/java/ch/blinkenlights/android/vanilla/PreferencesActivity.java]: I was thinking it doesn't really fit for the library screen, but you've got a point, it looks a bit strange. Will move.

**Comment by minus7** [comment]
I didn't add any translation strings besides English. I see you're using transifex but I don't know how I'd get the strings into that (or if I even have to)

File: app/src/main/java/ch/blinkenlights/android/vanilla/ShowQueueFragment.java (hunk 2 above)
Comment (minus7) [reply app/src/main/java/ch/blinkenlights/android/vanilla/ShowQueueFragment.java]: Done

**Comment by adrian-bl** [comment]
Only adding the english translation here is fine: Transifex pulls our english translation every ~24h, so it will automatically show up once this is merged.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+	<string name="scroll_to_current">Scroll to current song</string>
+</resources>
```
Comment (adrian-bl) [inline app/src/main/res/values/strings.xml]: This is not needed anymore, please revert

File: app/src/main/res/values/translatable.xml [hunk 4]
```diff
@@ -226,6 +226,8 @@ THE SOFTWARE.
 	<string name="cover_press_action_title">Cover tap action</string>
 	<string name="cover_longpress_action_title">Cover long press action</string>
 
+	<string name="queue_screen">Queue screen</string>
+
```
Comment (adrian-bl) [inline app/src/main/res/values/translatable.xml]: This is also not needed, please remove the line.

Will merge afterwards.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (minus7) [reply app/src/main/res/values/strings.xml]: Ah, kinda missed those two. Removed and squashed my commits into one

**Comment by adrian-bl** [comment]
looks good to me, thanks for the patch!

**Pull request merged**